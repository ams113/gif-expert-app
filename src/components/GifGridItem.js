import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { title = '', url = '' } ) => {

    const titleValid = title.trim() || 'Without title';

    return (
        <article className="card animate__animated animate__fadeIn">
            <div className="card-content">
                <img src={ url } alt={ titleValid } />
                <p>{ titleValid }</p>
            </div>
        </article>
    )
}


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  };