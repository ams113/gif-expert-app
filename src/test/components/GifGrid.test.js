import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Testing components: GifGrid', () => {
    
    const category = 'foo';
    
    test('should renders correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show items when loaded correctly', () => {

        const gifs = [
            {
                id: 'f00',
                title: 'foo',
                url: 'https://fake.dev/picture/fake.giff'
            },
            {
                id: 'f01',
                title: 'foo',
                url: 'https://fake.dev/picture/fake.giff'
            }
    ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
})
