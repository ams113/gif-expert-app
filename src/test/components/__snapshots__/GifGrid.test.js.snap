// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components: GifGrid should renders correctly 1`] = `
<div
  className="centered"
>
  <h3
    className="animate__animated animate__fadeIn"
  >
    foo
  </h3>
  <p
    className="animate__animated__flash"
  >
    Loading...
  </p>
  <section
    className="cards"
  />
</div>
`;

exports[`Testing components: GifGrid should show items when loaded correctly 1`] = `
<div
  className="centered"
>
  <h3
    className="animate__animated animate__fadeIn"
  >
    foo
  </h3>
  <section
    className="cards"
  >
    <GifGridItem
      id="f00"
      key="f00"
      title="foo"
      url="https://fake.dev/picture/fake.giff"
    />
    <GifGridItem
      id="f01"
      key="f01"
      title="foo"
      url="https://fake.dev/picture/fake.giff"
    />
  </section>
</div>
`;
